test/00_transform/MatMul.cu

```diff
@@ -325,6 +348,58 @@ TYPED_TEST(MatMulTestFloatTypes, MediumRectBatched3DStridedBatch)
   MATX_EXIT_HANDLER();
 }
 
+TYPED_TEST(MatMulTestFloatNonComplexTypes, MixedTypes)
+{
+  using ComplexTypeParam = typename float_to_complex_t<TypeParam>::t;
+  // a -> real, b -> real, c -> complex
+  {
+    MATX_ENTER_HANDLER();
+
+    constexpr index_t m = 4;
+    constexpr index_t k = 8;
+    constexpr index_t n = 16;
+
+    tensor_t<TypeParam, 2> a{{m, k}};
+    tensor_t<TypeParam, 2> b{{k, n}};
+    tensor_t<ComplexTypeParam, 2> c{{m, n}};
+
+    this->pb->template InitAndRunTVGenerator<TypeParam>(
+        "00_transforms", "matmul_operators", "run", {m, k, n});
+
+    this->pb->NumpyToTensorView(a, "a");
+    this->pb->NumpyToTensorView(b, "b");
+
+    matmul(c, a, b);
+    MATX_TEST_ASSERT_COMPARE(this->pb, c, "c", this->thresh);
+
+    MATX_EXIT_HANDLER();
+  }
+
+  // a -> real, b -> complex, c -> complex
+  {
+    MATX_ENTER_HANDLER();
+
+    constexpr index_t m = 4;
+    constexpr index_t k = 8;
+    constexpr index_t n = 16;
+
+    tensor_t<TypeParam, 2> a{{m, k}};
+    tensor_t<ComplexTypeParam, 2> b{{k, n}};
+    tensor_t<ComplexTypeParam, 2> c{{m, n}};
+
+    this->pb->template InitAndRunTVGenerator<TypeParam>(
```

Comment:
Without modifying the Python script, I'm not sure how this is working. This is using `TypeParam` (real) as the data type passed to Python. Inside of Python I think this is just doing a real * real, resulting in a real, but you are treating a real tensor as a complex. I think `b` has no imaginary values. Can you confirm?

Reply:
Yes. You are right. Would there be a simple way to check it? From what I see and understand `InitAndRunTVGenerator` always produces all np.arrays corresponding to TypeParam.

Thanks for catching this!

Reply:
Updated.